### PR TITLE
Address i18n issue with greeter timelabel.

### PR DESCRIPTION
**Pull request by vjr**
This is to address the first point in the issue https://github.com/elementary/greeter/issues/17 by combining the time and am/pm labels into a single text label format.

One nitpick I have is that the resulting AM/PM portion becomes larger where I preferred the current styling.

Note that I've removed the am/pm `meridien_format` label hence that string is removed from the code. Need to update the translations files which I have yet to figure out and add to this PR.

### CODE REVIEW

File: src/Widgets/TimeLabel.vala [hunk 1]
```diff
@@ -52,13 +47,10 @@ public class TimeLabel : Gtk.Grid {
         /// Date display, see http://valadoc.org/#!api=glib-2.0/GLib.DateTime.format for more details
         var day_format = _("%A, %B %e");
         /// Time display, see http://valadoc.org/#!api=glib-2.0/GLib.DateTime.format for more details
-        var time_format = _("%l:%M");
-        /// AM/PM display, see http://valadoc.org/#!api=glib-2.0/GLib.DateTime.format for more details. If you translate in a language that has no equivalent for AM/PM, keep the original english string.
-        var meridiem_format = _(" %p");
-
+        var time_format = Granite.DateTime.get_default_time_format (true);
```
Comment (matzipan) [inline src/Widgets/TimeLabel.vala]: Perhaps add a new line here.

**Comment by tintou** [comment]
Requested elementary/ux review because of the meridian label removal but it should be fine

**Comment by matzipan** [comment]
Ok so, testing on my machine, when I select another username in the list which has 24h instead of AM/PM set, the format doesn't switch.

**Comment by vjr** [comment]
> I don't know how much this is a styling error, this might be due to the AM/PM label getting to big as you mentioned, but the right hand side of the label is now exactly positioned against the right edge of the screen. Not sure where the change needs to take place.

Should be fixed in the latest commit https://github.com/elementary/greeter/pull/54/commits/05954ab7f300765938f10e65d2728bc793eed490

> Similarly - not entirely sure where this is coming from or if this is even related to this branch, but if I run ./pantheon-greeter or pantheon-greeter with the code on this branch, I get 24h format, but if I lock my screen I get the 12h format.

This may be because you have set your region to UK only in your own account but not in system-wide locale (in switchboard language-region settings) and I believe I also see this behaviour. This should be resolved with the other PR #53

**Comment by vjr** [comment]
Would anyone from @elementary/ux like to take a look and merge this approved PR?

**Comment by matzipan** [comment]
Hey @vjr can you action the last comment I made?

**Comment by vjr** [comment]
@matzipan apologies for the long delay - i will look into this and update (ASAP).

**Comment by cassidyjames** [comment]
Closing due to inactivity. Feel free to re-open if work picks back up!

**Pull request closed without merging**